components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/IdentityProviderSecretsProcessorImpl.java

```diff
@@ -0,0 +1,135 @@
+import org.wso2.carbon.identity.secret.mgt.core.model.SecretType;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.IDN_SECRET_TYPE_IDP_SECRETS;
+
+public class IdentityProviderSecretsProcessorImpl implements IdentityProviderSecretsProcessor {
+
+    private final SecretManager secretManager;
+    private final SecretResolveManager secretResolveManager;
+
+    public IdentityProviderSecretsProcessorImpl() {
+
+        this.secretManager = new SecretManagerImpl();
+        this.secretResolveManager = new SecretResolveManagerImpl();
+    }
+
+    @Override
+    public IdentityProvider addOrUpdateIdpSecrets(IdentityProvider identityProvider) throws SecretManagementException {
+
+        for (FederatedAuthenticatorConfig fedAuthConfig : identityProvider.getFederatedAuthenticatorConfigs()) {
+            for (Property prop : fedAuthConfig.getProperties()) {
+                if (prop.isConfidential()) {
+                    String secretName =
+                            buildSecretName(identityProvider.getId(), fedAuthConfig.getName(), prop.getName());
+                    if (secretManager.isSecretExist(IDN_SECRET_TYPE_IDP_SECRETS, secretName)) {
+                        // update existing secret
+                        updateExistingSecretProperty(secretName, prop);
+                        prop.setValue(buildSecretReference(secretName));
+                    } else {
+                        // add secret to the DB
+                        if (!StringUtils.isEmpty(prop.getValue())) {
+                            addNewIdpSecretProperty(secretName, prop);
+                            prop.setValue(buildSecretReference(secretName));
+                        }
+                    }
+                }
+            }
+        }
+
+        return identityProvider;
+    }
+
+    @Override
+    public IdentityProvider getIdpSecrets(IdentityProvider identityProvider)
+            throws SecretManagementException {
+
+        for (FederatedAuthenticatorConfig fedAuthConfig : identityProvider.getFederatedAuthenticatorConfigs()) {
+            for (Property prop : fedAuthConfig.getProperties()) {
+                if (prop.isConfidential()) {
+                    String secretName =
+                            buildSecretName(identityProvider.getId(), fedAuthConfig.getName(), prop.getName());
+                    if (secretManager.isSecretExist(IDN_SECRET_TYPE_IDP_SECRETS, secretName)) {
+                        ResolvedSecret resolvedSecret =
+                                secretResolveManager.getResolvedSecret(IDN_SECRET_TYPE_IDP_SECRETS, secretName);
+                        // replace secret reference with decrypted original secret
+                        prop.setValue(resolvedSecret.getResolvedSecretValue());
+                    }
+                }
+            }
+        }
+
+        return identityProvider;
+    }
+
+    @Override
+    public IdentityProvider deleteIdpSecrets(IdentityProvider identityProvider)
```

Comment:
Make it return void

Reply:
Resolved with [039da67](https://github.com/wso2/carbon-identity-framework/pull/4430/commits/039da6750bc84e5e950df73f8c057a0cf6567a87)